### ISSUE TITLE
Feature Request: Project Folders

### ISSUE THREAD

**Issue by denysvitali** [hashicorp/waypoint#1765]
**Is your feature request related to a problem? Please describe.**
Initially I though that the Project -> Application structure was basically a way to group multiple applications (e.g: applications part of a project) into a single container. This unfortunately is not really the case as per Waypoint terms, a project is a single deployable entity.

**Describe the solution you'd like**
To better separate macro-entities from the single deployable things (in this case, `waypoint.hcl` files), one might want to group them in a way that it makes sense then to easily navigate the UI.

For example, at Google one might create the "Google Drive" group and then associate to that specific group N projects, for example `google-drive-backend`, `google-drive-android-app`, `google-drive-web`.

In Waypoint, this macro categorization doesn't seem to exist and therefore the UI (right from the home page) will show all three projects mentioned above, listed there. You can totally see how this will grow to a tremendous list of projects that you'll have to `CTRL+F` to find.

**Describe alternatives you've considered**
I guess that a way to tag projects or add them to hierarchical folders would be enough to solve the issue

**Explain any additional use-cases**
I don't think there are additional use cases

**Additional context**
N/A

**Comment by krantzinator** [comment]
The architecture around this is currently being discussed internally. Thanks for this input!

**Comment by denysvitali** [comment]
(is there a way you can maybe set-up some calls / public discussions about these things with the community so that we can better understand the direction Waypoint is taking?)

**Comment by krantzinator** [comment]
This is absolutely open for discussion! Any additional input you or other contributors want to share here is welcome. I think I was a bit too brief in my original response, so let me try again 🙂 
When I first joined the Waypoint project, my perception of the intent of Projects matched the understanding you shared above, so I share your desire to iron this out. I think your idea of adding delineations to the UI is a useful one, but I would want the concept to be shared by the CLI as well. Currently, our "one project, one git repo" mapping is a pretty foundational one.
As I write that out, I realize your concern may be mainly with how those relationships are presented in the UI, rather than with the underlying structure?

**Comment by denysvitali** [comment]
> As I write that out, I realize your concern may be mainly with how those relationships are presented in the UI, rather than with the underlying structure?

For this specific issue, yes, you're right, purely organizational / UI.

> Currently, our "one project, one git repo" mapping is a pretty foundational one.

This is a paradigm that I'm challenging over at #1796: this paradigm is good for many use cases, but it's completely off for our case. I guess that this part is better discussed in that other issue :+1:

**Comment by krantzinator** [comment]
Thanks for the clarification! Will check out that other issue as well.
